### PR TITLE
Add all IANA assigned HTTP method names

### PR DESCRIPTION
**Pull request by abhinavsingh**
Additionally:
- Fixes the additional `v` prefix in printed version
- Now every `make` invocation regenerates `proxy/common/_scm_version.py` for latest version string

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -26,4 +26,6 @@ cover
 htmlcov
 dist
 build
+
 proxy/public
+docs/_build
```
Comment (webknjaz) [inline .gitignore:31]: Why did you do this? It seems like you're duplicating tox's effort. It's a bad idea.

File: Makefile [hunk 2]
```diff
@@ -82,14 +83,13 @@ lib-clean:
...
 	pip install \
 		-r requirements.txt \
 		-r requirements-testing.txt \
 		-r requirements-release.txt \
-		-r requirements-tunnel.txt
-
-lib-scm-version:
-	@echo "version = '$(VERSION)'" > proxy/common/_scm_version.py
+		-r requirements-tunnel.txt \
+		-r docs/requirements.txt
```
Comment (webknjaz) [inline Makefile:92]: It's a bad idea to mix up test and docs requirements. They should be separate.

File: Makefile [hunk 3]
```diff
@@ -114,6 +114,16 @@ lib-release-test: lib-package
 lib-release: lib-package
 	twine upload dist/*
 
+lib-doc:
+	pushd docs && \
+	python -m sphinx \
```
Comment (webknjaz) [inline Makefile:119]: No need to maintain a separate copy of the build process. Call `tox` like everybody else in the ecosystem. The docs deps are conceptually different from the test ones. By mixing them up you increase the chance of hitting pointless dependency conflicts.
Just do `tox -e build-docs`.

File: tests/http/test_http_parser.py [hunk 4]
```diff
@@ -24,10 +24,10 @@ def setUp(self) -> None:
         self.parser = HttpParser(httpParserTypes.REQUEST_PARSER)
 
     def test_issue_127(self) -> None:
-        with self.assertRaises(NotImplementedError):
+        with self.assertRaises(ValueError):
```
Comment (webknjaz) [inline tests/http/test_http_parser.py:27]: These CMs like `pytest.raises()`/`pytext.warns()` usually accept the second argument for the message. It is the best practice to always pass that for precise exception expectations.

File: write-scm-version.sh [hunk 5]
```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
+#
+# write-scm-version.sh exists because `proxy.py``
+# auto-detects it's next version from git.  Hence,
+# for `proxy.py` to work `proxy/common/_scm_version.py`
+# file must be auto-generated with necessary information.
+#
+# For CI/CD, this file is generated via `tox` integration.
+# For local development (without editable install), you
```
Comment (webknjaz) [inline write-scm-version.sh:9]: I'd argue, that for local development, an editable install must be used which would take care of this w/o the need for this script.

File: .gitignore (hunk 1 above)
Comment (abhinavsingh) [reply .gitignore:31]: I used `python -m sphinx` and I ended up with `docs/_build`.  May be we should update that to use `tox`

File: Makefile (hunk 2 above)
Comment (abhinavsingh) [reply Makefile:92]: Agreed.  This should be part of `make lib-doc` itself

File: tests/http/test_http_parser.py (hunk 4 above)
Comment (abhinavsingh) [reply tests/http/test_http_parser.py:27]: Agreed, this needs to be covered across all `raise` expectations I guess.

File: write-scm-version.sh (hunk 5 above)
Comment (abhinavsingh) [reply write-scm-version.sh:9]: Agreed.  I'll most likely remove this bootstrapping once readme is updated with editable install as go-to developer setup step.

File: .gitignore (hunk 1 above)
Comment (webknjaz) [reply .gitignore:31]: Well, the idea is that you don't do this because tox already has that wrapped and managed under `.tox/`.

File: Makefile [hunk 6]
```diff
@@ -114,6 +114,16 @@ lib-release-test: lib-package
...
+		--keep-going \
+		-b dirhtml \
+		-d _build/doctrees \
+		-D language=en . _build/html && \
+	popd && \
+	open docs/_build/html/index.html
```
Comment (webknjaz) [inline Makefile:125]: Also, `open` only exists on macOS. Here you exclude all the Linux-based contributors who'd need an `xdg-open.

File: Makefile (hunk 3 above)
Comment (webknjaz) [reply Makefile:119]: https://github.com/abhinavsingh/proxy.py/pull/752

File: Makefile (hunk 6 above)
Comment (abhinavsingh) [reply Makefile:125]: Thank you, will consider presence of `xdg-open` or add a OS check

**Pull request merged**